tests_integration/test_stage.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+
+import os
+from unittest import mock
+from tempfile import NamedTemporaryFile
+from tests_integration.snowflake_connector import snowflake_session
+from tests_integration.test_utils import (
+    extract_row_from_mock,
+    extract_row_from_snowflake_session,
+    contains_row_with,
+    not_contains_row_with,
+)
+
+# TODO add database when we will have one
+DATABASE_NAME = ""
+
+
+@pytest.mark.integration
+@mock.patch("snowcli.cli.stage.print_db_cursor")
+def test_stage(mock_print, runner, snowflake_session, tmp_path):
+    stage_name = "test_stage"
+
+    runner.invoke_with_config(["stage", "create", stage_name])
+    assert contains_row_with(
+        extract_row_from_mock(mock_print),
+        {"status": f"Stage area {stage_name.upper()} successfully created."},
+    )
+
+    runner.invoke_with_config(["stage", "list"])
+    expect = snowflake_session.execute_string(
+        f"show stages like '{stage_name}' in database {DATABASE_NAME}"
+    )
+    assert contains_row_with(
+        extract_row_from_mock(mock_print), extract_row_from_snowflake_session(expect)[0]
+    )
+
+    with NamedTemporaryFile("w+", suffix=".txt") as tmp_file:
+        runner.invoke_with_config(["stage", "put", tmp_file.name, stage_name])
+        filename = os.path.basename(tmp_file.name)
+        assert contains_row_with(
+            extract_row_from_mock(mock_print),
+            {"source": filename, "target": filename, "status": "UPLOADED"},
+        )
+
+    runner.invoke_with_config(["stage", "list", stage_name])
+    result = snowflake_session.execute_string(
+        f"use database {DATABASE_NAME}; list @{stage_name}"
+    )
+    assert extract_row_from_mock(mock_print) == extract_row_from_snowflake_session(
```

Comment:
Maybe we can dump the `extract` prefix, wdyt?

Reply:
I agree 👍